src/main/java/application/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,172 @@
+package application.dao.jdbc;
+
+import application.dao.ShoppingCartDao;
+import application.exceptions.DataProcessingException;
+import application.lib.Dao;
+import application.model.Product;
+import application.model.ShoppingCart;
+import application.util.ConnectionUtil;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ShoppingCartDaoJdbcImpl implements ShoppingCartDao {
+    @Override
+    public Optional<ShoppingCart> getByUserId(Long userId) {
+        ShoppingCart shoppingCart = null;
+        String query = "SELECT * FROM shopping_carts WHERE user_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, userId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                shoppingCart = getCartFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get shopping cart for user with id "
+                    + userId, e);
+        }
+        shoppingCart.setProducts(getProducts(shoppingCart.getId()));
+        return Optional.of(shoppingCart);
```

Comment:
you will have NPE on shoppingCart.setProducts() 
add check and return optional empty if cart was not found

Reply:
done